omod_1.x/src/main/java/org/openmrs/module/openhmis/cashier/web/controller/JasperReportController.java

```diff
@@ -59,7 +59,11 @@ public String render(@RequestParam(value = "reportId", required = true) int repo
 		HashMap<String, Object> params = new HashMap<String, Object>();
 		params.put("timesheetId", timesheetId);
 		try {
-			ReportGenerator.generate(report, params, false, true);
+			if (Context.getAuthenticatedUser() == null) {
```

Comment:
This check should be done as the first part of the render method. See the similar issue for the inventory pull request.